wallet/wallet-cli-lib/src/lib.rs

```diff
@@ -100,30 +102,30 @@ pub async fn run(
         .await
         .map_err(WalletCliError::RpcError)?;
 
-    let mut controller_opt = None;
+    let controller_opt = None;
+
+    let (event_tx, event_rx) = mpsc::unbounded_channel();
 
     if let Some(wallet_path) = wallet_file {
-        commands::handle_wallet_command(
-            &chain_config,
-            &rpc_client,
-            &mut controller_opt,
-            commands::WalletCommand::OpenWallet { wallet_path },
-        )
-        .await?;
+        let (res_tx, _res_rx) = tokio::sync::oneshot::channel();
+        event_tx
+            .send(Event::HandleCommand {
+                command: WalletCommand::OpenWallet { wallet_path },
+                res_tx,
+            })
+            .expect("should not fail");
```

Comment:
Should there be more context here e.g. `.expect("Opening a wallet should not fail")`

Reply:
The except is actually for sending the event through the channel.